WordPress/wporg-mu-plugins
# Tools: Update packages, use wp-scripts for build process

The Interactivity API is now only available as a JS module, not a plain script, so we need to build and register our `view` code differently. As of v27, `@wordpress/scripts` supports module builds, but that also breaks compat with Node <18. So this PR updates the [node version to match core](https://make.wordpress.org/core/2023/12/20/updating-wordpress-to-use-more-modern-versions-of-node-js-npm-2/), and updates the rest of the JS packages.

I've also changed the build process to call wp-scripts directly for each block, rather than using webpack + importing the wp-scripts webpack config. The block build step is now equivalent to running:

```bash
npx wp-scripts build --experimental-modules --webpack-src-dir=./mu-plugins/blocks/[block]/src --output-path=./mu-plugins/blocks/[block]/build
```

Unfortunately due to the folder structure, we still need the custom script (wp-scripts doesn't understand how to find `[block]/src/block.json`, it only understands `src/[block]/block.json`).

See https://github.com/WordPress/wporg-parent-2021/issues/122, https://github.com/WordPress/gutenberg/pull/56143, https://github.com/WordPress/gutenberg/pull/57461

**To test**

- Run `nvm install` to switch the node version (or however you manage node versions), it should be 20.
- Run `yarn` to install the package updates
    - You might need to remove node_modules first, I had a strange issue with wp-scripts not updating correctly
- Run `yarn build`
    - All builds should succeed, both CSS and JS
    - View any file, it should be minified, ex: `cat mu-plugins/blocks/navigation/build/index.js`
- Run the new `yarn build:dev`
    - All builds should succeed
    - View any file, it should not be minified, ex: `cat mu-plugins/blocks/navigation/build/index.js`
- Run the build steps with a block, ex `yarn build:dev query-filter`
- Run the watch script
    - The initial build of all modules should succeed
    - Changing any files in blocks, JS, CSS, etc, should trigger a rebuild of the files
